riscv/tests/riscv_data/continuation.rs

```diff
@@ -0,0 +1,51 @@
+#![no_std]
+
+extern crate alloc;
+
+use runtime::coprocessors::poseidon_gl_unsafe;
+use runtime::{get_prover_input, print};
+
+// 256 words = 1KB
+static PAGE_SIZE: usize = 256;
+
+#[no_mangle]
+pub fn main() {
+    let num_pages = get_prover_input(0) as usize;
+    for page in 0..num_pages {
+        let page_start_index = (page * (PAGE_SIZE + 1) + 1) as u32;
+        let start_address = get_prover_input(page_start_index) as usize;
+
+        // Hash buffer to linearly hash the page.
+        // The 12 elements consist of 3 parts:
+        // - The first 4 elements are current 4 words
+        // - The next 4 elements are the previous hash
+        // - The last 4 elements are the capacity elements (always 0)
```

Comment:
Actually, I'm pretty sure that it would be secure if we'd put the previous hash into the last 4 elements (and 0 at the first). Pretty sure that's how Polygon hashes strings. That would cut the number of hashes in half.